src/google/protobuf/compiler/java/full/enum.cc

```diff
@@ -263,7 +263,7 @@ void EnumNonLiteGenerator::Generate(io::Printer* printer) {
         "    getDescriptorForType() {\n"
         "  return getDescriptor();\n"
         "}\n"
-        "public static final com.google.protobuf.Descriptors.EnumDescriptor\n"
+        "public static com.google.protobuf.Descriptors.EnumDescriptor\n"
```

Comment:
I actually think this one is not really redundant either, because it still has the effect of prevent subclasses from hiding this method.

Reply:
This is inside the enum so it can't be subclassed right?

Reply:
Thanks! I know it's a minor change, but I noticed it and saw an opportunity to contribute. 🙂